eras/conway/impl/src/Cardano/Ledger/Conway/Governance.hs

```diff
@@ -593,69 +597,156 @@ pparamsUpdateThreshold pp ppu =
         Set.map lookupGroupThreshold $
           modifiedGroups @era ppu
 
-thresholdSPO ::
+data VotingThreshold
+  = -- | This is the actual threshold. It is lazy, because upon proposal we only care if
+    -- the voting is allowed or not, instead of getting the actual threshold value.
+    VotingThreshold UnitInterval -- <- lazy on purpose
+  | -- | Does not have a threshold, therefore an action can not be ratified
+    NoVotingThreshold
+  | -- | Some GovActions are not alloweed to be voted by some entities
+    NoVotingAllowed
+
+toRatifyVotingThreshold :: VotingThreshold -> StrictMaybe UnitInterval
+toRatifyVotingThreshold = \case
+  VotingThreshold t -> SJust t -- concrete threshold
+  NoVotingThreshold -> SNothing -- no voting threshold prevents ratification
+  NoVotingAllowed -> SJust minBound -- votes should not count, set threshold to zero
```

Comment:
The naming is a bit confusing. I would've thought that `NoVotingThreshold` means that it's automatically accepted and `NoVotingAllowed` means it never passes

Reply:
Maybe it's better to rename to `AlwaysPass` and `NeverPass`

Reply:
> I would've thought that NoVotingThreshold means that it's automatically accepted and NoVotingAllowed means it never passes

That's because it is not. It is the opposite.

* `NoVotingThreshold` -> it will never be accepted
* `NoVotingAllowed` -> it will always be accepted.

Problem with naming is that it is not only about accepting governance actions, but it is also about allowing for them to be voted on. 

If I name it:
* `NeverAccept` or `NeverPass`
* `AlwaysAccept` or `AlwaysAccept`

Then why `AlwaysPass` does not allow voting and `NeverPass` does allow voting?

It is a bit confusing, but I can't think of a better naming. Any other ideas?

Reply:
We can just as well prevent voting on both `AlwaysPass` and `NeverPass`. Returning `SNothing` as the threshold essentially means that `threshold > 1`, it doesn't have anything to do with preventing voting.

Reply:
I also just realized that `VotingThreshold minBound` and `NoVotingAllowed` are essentially the same thing

Reply:
Oh, ok, except for `isVotingAllowed`

Reply:
Maybe it's a good idea to somehow prevent constructing `VotingThreshold minBound`

Reply:
> Oh, ok, except for isVotingAllowed

That's the whole point of this PR. This function is necessary for #3718 

> Maybe it's a good idea to somehow prevent constructing VotingThreshold minBound

`VotingThreshold minBound` is a legitimate value, because nothing says any of those thresholds can't be set to 0 in the PParams.

Reply:
> We can just as well prevent voting on both AlwaysPass and NeverPass

We can't. Those are two separate concepts:

* For one of them (`NoVotingAllowed`) voting is not counted by RATIFY rule and will be prevented in GOV rule in #3718
* For the other (`NoVotingThreshold`) we do allow votes, but we can't RATIFY them. Which makes sense for:
  * InfoAction
  * Committee votes when there is no committee. Note that, committee members can still submit votes, even if there is currently no committee, because a committee member might now that he'll be elected soon. Once he's elected his votes will be counted. 
  
So, for `NoVotingThreshold` we can't disallow voting, since that would prohibit those use cases.